netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -248,7 +250,16 @@ public void operationComplete(ChannelFuture future) throws Exception {
       keepAliveManager.onTransportStarted();
     }
 
-    return null;
+    return new Runnable() {
+      @Override
+      public void run() {
+        try {
+          fenceOperation.await();
```

Comment:
This is still racey: we will need to block until the *listener* is done, not the future that the listener is waiting on. Update coming soon.